content/guides/advanced/testing-with-jest-and-bankrun.md

```diff
@@ -0,0 +1,380 @@
+   the environment variables set.
+
+## Bankrun
+
+Instead of using `solana-test-validator` you can also use
+[Solana Bankrun](https://kevinheavey.github.io/solana-bankrun/). It acts
+similarly to the local validator but is more lightweight and faster. Some really
+helpful features are
+[writing custom account data](https://kevinheavey.github.io/solana-bankrun/tutorial/#writing-arbitrary-accounts)
+and
+[time travel](https://kevinheavey.github.io/solana-bankrun/tutorial/#time-travel)
+which makes it much easier to test time based programs and programs that rely on
+specific account data.
+
+To use Bankrun and Bankrun Anchor you need to add it to your `package.json`:
+
+```bash
+yarn add solana-bankrun anchor-bankrun
+```
+
+To switch from a Mocha Anchor test to a
+[Bankrun Anchor](https://github.com/kevinheavey/anchor-bankrun) test you only
+need to change the provider to be a
+[`BankrunProvider`](https://kevinheavey.github.io/solana-bankrun/tutorial/#anchor-integration)
+and create a context using `startAnchor`:
+
+```js
+const context = await startAnchor(".", [], []);
+const client = context.banksClient;
+
+const provider = new BankrunProvider(context);
+anchor.setProvider(provider);
+```
+
+`startAnchor` will automatically add your program from your Anchor workspace to
+the Bankrun bank. You can also
+[add additional accounts and programs](https://kevinheavey.github.io/solana-bankrun/tutorial/#writing-arbitrary-accounts)
+to the bank by passing them in the `startAnchor` function. There are a few
+things that are different to running tests against the local validator though
+which we will cover in the next section.
+
+### Bankrun differences to the local validator
+
+Because Bankrun directly runs your tests against a local Bank the tests are
+faster, but there are a few differences to the local validator:
+
+#### Airdrops
+
+- Bankrun does not support Airdrops. The standard signer used in the
+  BankrunProvider will be automatically funded with some Sol. If you need
+  another funded account you can create one by passing in an additional account
+  in the `startAnchor` function.
+
+```js
+let secondKeypair: Keypair = new anchor.web3.Keypair();
+
+context = await startAnchor(
+    "",[],
+    [
+    {
+        address: secondKeypair.publicKey,
+        info: {
+        lamports: 1_000_000_000, // 1 SOL equivalent
+        data: Buffer.alloc(0),
+        owner: SYSTEM_PROGRAM_ID,
+        executable: false,
+        },
+    },
+    ]
+);
+provider = new BankrunProvider(context);
+```
+
+#### Confirming transactions
+
+Since Bankrun is directly working on the bank you will not need to confirm your
+transactions. So the `connection.confirmTransaction()` function will not be
+available. You can just leave it out.
+
+#### Getting account data
+
+While you can still use `connection.getAccount` to get the account data of an
+account you can also use the `client.getAccount` function to get the account
```

Comment:
whats the preferred way to get data? we should recommend which ever is better/faster